### PR TITLE
fix: Kafka Auth Mechanism "SERVER_ROOT_CA_CERTIFICATE" and other Auth can't be used together

### PR DESCRIPTION
**Pull request by ConnorRobertson**
### Issue #, if available
[#3339](https://github.com/aws/serverless-application-model/issues/3339)
### Description of changes
Removed `"SERVER_ROOT_CA_CERTIFICATE"` from `AUTH_MECHANISM` as unlike the other types in `AUTH_MECHANISM` `"SERVER_ROOT_CA_CERTIFICATE"` can be enabled at the same time.

### Checklist

- [ ] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by aaythapa — COMMENTED** [review]
some transform tests?

**Comment by ConnorRobertson** [comment]
> some transform tests?

Will be added just want to confirm everything is as intended first.

File: samtranslator/model/eventsources/pull.py [hunk 1]
```diff
@@ -620,7 +628,7 @@ def get_secret_key(self, source_access_configurations: List[Any]):  # type: igno
                 self.relative_id,
                 "VPC_SUBNET and VPC_SECURITY_GROUP in SourceAccessConfigurations for SelfManagedKafka must be both provided.",
             )
-        return authentication_uri, (has_vpc_subnet and has_vpc_security_group)
+        return authentication_uri, authentication_uri_2, (has_vpc_subnet and has_vpc_security_group)
```
Comment (ssenchenko) [inline samtranslator/model/eventsources/pull.py:631]: Since we're modifying the return, let's update the typings

**Review by ssenchenko — APPROVED** [review]

**Review by GavinZZ — COMMENTED** [review]
Let's sync on this tmr. I'm aa bit confused on this issue.

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**